cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -165,12 +166,17 @@ class TestBinaryArithmetic : public TestBase {
   void ValidateAndAssertApproxEqual(const std::shared_ptr<Array>& actual,
                                     const std::shared_ptr<Array>& expected) {
     ASSERT_OK(actual->ValidateFull());
-    AssertArraysApproxEqual(*expected, *actual, /*verbose=*/true);
+    AssertArraysApproxEqual(*expected, *actual, /*verbose=*/true, equal_options_);
   }
 
   void SetOverflowCheck(bool value = true) { options_.check_overflow = value; }
 
+  void SetNansEqual(bool value = false) {
```

Comment:
This is unnecessary, just always set it to true.

Reply:
Also, can you test `Nan` in the inputs in `TestBinaryArithmeticFloating` now?

Reply:
Sounds reasonable. 
I have made the default value to true, and provided a test case for NaN in the input.